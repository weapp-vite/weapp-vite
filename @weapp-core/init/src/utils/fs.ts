import fs from 'fs-extra'

export async function readJsonIfExists<T>(filepath: string): Promise<T | null> {
  try {
    if (!await fs.pathExists(filepath)) {
      return null
    }
    return await fs.readJSON(filepath) as T
  }
  catch (error) {
    throw new FsReadError(filepath, error)
  }
}

export async function readFileIfExists(filepath: string): Promise<string | null> {
  try {
    if (!await fs.pathExists(filepath)) {
      return null
    }
    return await fs.readFile(filepath, 'utf8')
  }
  catch (error) {
    throw new FsReadError(filepath, error)
  }
}

export async function writeJsonFile(filepath: string, data: unknown, spaces = 2) {
  try {
    await fs.outputJSON(filepath, data, {
      spaces,
    })
  }
  catch (error) {
    throw new FsWriteError(filepath, error)
  }
}

export async function writeFile(filepath: string, contents: string) {
  try {
    await fs.outputFile(filepath, contents, 'utf8')
  }
  catch (error) {
    throw new FsWriteError(filepath, error)
  }
}

export class FsReadError extends Error {
  constructor(public filepath: string, public cause: unknown) {
    super(`Failed to read ${filepath}`)
    this.name = 'FsReadError'
  }
}

export class FsWriteError extends Error {
  constructor(public filepath: string, public cause: unknown) {
    super(`Failed to write ${filepath}`)
    this.name = 'FsWriteError'
  }
}
