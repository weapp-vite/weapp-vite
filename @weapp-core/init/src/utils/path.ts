import path from 'pathe'

export function resolveOutputPath(root: string, dest: string | undefined, fallback: string) {
  if (!dest) {
    return fallback
  }

  return path.isAbsolute(dest) ? dest : path.resolve(root, dest)
}
