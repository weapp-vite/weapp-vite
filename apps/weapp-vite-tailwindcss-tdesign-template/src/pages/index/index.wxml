<view class="min-h-screen {{ mode === 'light'?'bg-gray-100 text-slate-800':'bg-gray-900 text-slate-200' }} transition-colors duration-500">
  <view class="flex flex-col items-center pt-20 space-y-10">
    <HelloWorld></HelloWorld>
    <view bind:tap="switchMode" class="flex items-center">
      当前模式 {{mode}} 切换模式
      <view class="{{ mode === 'light' ? 'i-mdi-moon-waxing-crescent':'i-mdi-weather-sunny text-white' }} text-2xl"></view>
    </view>
    <view class="underline" mark:url="https://vite.icebreaker.top" bind:tap="copy">
      https://vite.icebreaker.top
    </view>
    <view class="underline" mark:url="https://tw.icebreaker.top" bind:tap="copy">
      https://tw.icebreaker.top
    </view>
    <view>
      <t-button theme="primary" ghost bind:tap="handleAction">TDesign Button</t-button>
    </view>
  </view>
  <t-action-sheet id="t-action-sheet" bind:selected="handleSelected" />
</view>
