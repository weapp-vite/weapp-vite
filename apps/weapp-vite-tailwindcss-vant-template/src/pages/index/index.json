{
  "$schema": "https://vite.icebreaker.top/page.json",
  "usingComponents": {
    "HelloWorld": "/components/HelloWorld/HelloWorld"
  },
  "navigationBarTitleText": "初始模板"
}
