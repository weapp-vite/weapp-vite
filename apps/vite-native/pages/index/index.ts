import Toast from '@vant/weapp/toast/toast'
// index.js
// import { formatTime } from '../../utils/util'
import logoUrl from '@/assets/logo.png'
// const { formatTime } = require('../../utils/util')
import { formatTime } from '../../utils/util'

require
  .async('./what/the')
  .then((mod) => {
    console.log(mod)
  })
  .catch(({ errMsg, mod }) => {
    console.error(`path: ${mod}, ${errMsg}`)
  })

let common
require('./what/you.js', (mod) => {
  common = mod
  console.log(common)
}, ({ errMsg, mod }) => {
  console.error(`path: ${mod}, ${errMsg}`)
})
console.log(common)

console.log('-------------', import.meta.env, import.meta.env.VITE_XXX)
const { PLATFORM } = import.meta.env
console.log(PLATFORM)
const defaultAvatarUrl = 'https://mmbiz.qpic.cn/mmbiz/icTdbqWNOwNRna42FI242Lcia07jQodd2FJGIYQfG0LAJGFxM4FbnQP6yfMxBgJ0F3YRqJCJ1aPAK2dQagdusBZg/0'
Page({
  data: {
    logoUrl,
    motto: JSON.stringify(import.meta.env, null, 2),
    message: 'Hello MINA!',
    userInfo: {
      avatarUrl: defaultAvatarUrl,
      nickName: '',
    },
    array: [1, 2, 3, 4, 5],
    hasUserInfo: false,
    canIUseGetUserProfile: wx.canIUse('getUserProfile'),
    canIUseNicknameComp: wx.canIUse('input.type.nickname'),
    time: formatTime(new Date()),
    staffA: { firstName: 'Hulk', lastName: 'Hu' },
    staffB: { firstName: 'Shang', lastName: 'You' },
    staffC: { firstName: 'Gideon', lastName: 'Lin' },
    flag: true,
  },
  bindViewTap() {
    wx.navigateTo({
      url: '../logs/logs',
    })
  },
  switchTemplate(e) {
    console.log(this.data.flag)
    this.setData({
      flag: !this.data.flag,
    })
  },
  onChooseAvatar(e) {
    const { avatarUrl } = e.detail
    const { nickName } = this.data.userInfo
    this.setData({
      'userInfo.avatarUrl': avatarUrl,
      'hasUserInfo': nickName && avatarUrl && avatarUrl !== defaultAvatarUrl,
    })
  },
  onInputChange(e) {
    const nickName = e.detail.value
    const { avatarUrl } = this.data.userInfo
    this.setData({
      'userInfo.nickName': nickName,
      'hasUserInfo': nickName && avatarUrl && avatarUrl !== defaultAvatarUrl,
    })
  },
  getUserProfile() {
    // 推荐使用wx.getUserProfile获取用户信息，开发者每次通过该接口获取用户个人信息均需用户确认，开发者妥善保管用户快速填写的头像昵称，避免重复弹窗
    wx.getUserProfile({
      desc: '展示用户信息', // 声明获取用户个人信息后的用途，后续会展示在弹窗中，请谨慎填写
      success: (res) => {
        console.log(res)
        this.setData({
          userInfo: res.userInfo,
          hasUserInfo: true,
        })
      },
    })
  },
  async hello(...args: any[]) {
    console.log(...args)

    const res = await import('./what/fuck')
    // 被转化成
    // const res = await Promise.resolve().then(() => require("../../fuck-D0aOZbBZ.js"));
    console.log(res)
    Toast('我是提示文案，建议不超过十五字~')
  },
})
