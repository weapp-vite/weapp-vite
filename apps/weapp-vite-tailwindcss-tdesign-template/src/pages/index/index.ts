import ActionSheet, { ActionSheetTheme } from 'tdesign-miniprogram/action-sheet/index'
import { hello } from '@/utils/util'

const firstGrid = [
  {
    label: '微信',
    image: 'https://tdesign.gtimg.com/mobile/demos/wechat.png',
  },
  {
    label: '朋友圈',
    image: 'https://tdesign.gtimg.com/mobile/demos/times.png',
  },
  {
    label: 'QQ',
    image: 'https://tdesign.gtimg.com/mobile/demos/qq.png',
  },
  {
    label: '企业微信',
    image: 'https://tdesign.gtimg.com/mobile/demos/wecom.png',
  },
  {
    label: '收藏',
    icon: 'star',
  },
  {
    label: '刷新',
    icon: 'refresh',
  },
  {
    label: '下载',
    icon: 'download',
  },
  {
    label: '复制',
    icon: 'queue',
  },
]
Page({
  data: {
    mode: 'light',
    hello: {
      title: 'Hello weapp-vite + TDesign',
      description: '集成 TDesign Miniprogram 与 Tailwind CSS，帮助你快速搭建企业级交互界面。',
      docs: 'https://vite.icebreaker.top',
      links: [
        {
          text: '复制文档链接',
          url: 'https://vite.icebreaker.top',
        },
        {
          text: 'TDesign 小程序组件',
          url: 'https://tdesign.tencent.com/miniprogram/overview',
          variant: 'ghost',
        },
      ],
    },
  },
  switchMode() {
    if (this.data.mode === 'light') {
      this.setData({
        mode: 'dark',
      })
    }
    else {
      this.setData({
        mode: 'light',
      })
    }
  },
  async copy(e: WechatMiniprogram.BaseEvent) {
    if (e.mark?.url) {
      await wx.setClipboardData({
        data: e.mark.url,
      })
      console.log(`复制成功: ${e.mark.url}`)
    }
  },
  handleSelected(e: WechatMiniprogram.CustomEvent) {
    console.log(e.detail)
  },
  onLoad() {
    console.log(hello())
  },
  handleAction() {
    ActionSheet.show({
      theme: ActionSheetTheme.Grid,
      selector: '#t-action-sheet',
      context: this,
      items: firstGrid,
      align: 'center',
      description: '',
    })
  },
})
