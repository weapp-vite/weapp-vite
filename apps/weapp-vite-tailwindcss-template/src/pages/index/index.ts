import { hello } from '@/utils/util'

Page({
  data: {
    mode: 'light',
    hello: {
      title: 'Hello weapp-vite',
      description: '这是一个集成 Tailwind CSS 的小程序模板，开箱即可体验原子化样式与完整的 HMR 流程。',
      docs: 'https://vite.icebreaker.top',
      links: [
        {
          text: '复制文档链接',
          url: 'https://vite.icebreaker.top',
        },
        {
          text: 'Tailwind Demo',
          url: 'https://tw.icebreaker.top',
          variant: 'ghost',
        },
      ],
    },
  },
  switchMode() {
    if (this.data.mode === 'light') {
      this.setData({
        mode: 'dark',
      })
    }
    else {
      this.setData({
        mode: 'light',
      })
    }
  },
  async copy(e: WechatMiniprogram.BaseEvent) {
    if (e.mark?.url) {
      await wx.setClipboardData({
        data: e.mark.url,
      })
      console.log(`复制成功: ${e.mark.url}`)
    }
  },
  onLoad() {
    console.log(hello())
  },
})
