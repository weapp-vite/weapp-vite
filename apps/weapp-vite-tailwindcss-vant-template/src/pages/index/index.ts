import Dialog from '@vant/weapp/dialog/dialog'
import { hello } from '@/utils/util'

Page({
  data: {
    mode: 'light',
    hello: {
      title: 'Hello weapp-vite + Vant',
      description: '基于 weapp-vite 与 Tailwind CSS，并预先集成了 Vant Weapp 组件库，助你快速搭建业务界面。',
      docs: 'https://vite.icebreaker.top',
      links: [
        {
          text: '复制文档链接',
          url: 'https://vite.icebreaker.top',
        },
        {
          text: 'Vant 组件手册',
          url: 'https://vant-contrib.gitee.io/vant-weapp/#/home',
          variant: 'ghost',
        },
      ],
    },
  },
  switchMode() {
    if (this.data.mode === 'light') {
      this.setData({
        mode: 'dark',
      })
    }
    else {
      this.setData({
        mode: 'light',
      })
    }
  },
  async copy(e: WechatMiniprogram.BaseEvent) {
    if (e.mark?.url) {
      await wx.setClipboardData({
        data: e.mark.url,
      })
      console.log(`复制成功: ${e.mark.url}`)
    }
  },
  showDialog() {
    Dialog.alert({
      title: '我来自Vant',
      message: '我是Dialog',
    }).then(() => {
      // on close
    })
  },
  onLoad() {
    console.log(hello())
  },
})
