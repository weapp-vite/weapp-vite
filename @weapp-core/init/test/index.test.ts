import CI from 'ci-info'
import { omit } from 'es-toolkit/compat'
import fs from 'fs-extra'
import path from 'pathe'
import { createOrUpdatePackageJson, createOrUpdateProjectConfig, initConfig, initViteConfigFile } from '@/index'

const appsDir = path.resolve(__dirname, '../../../apps')
const fixturesDir = path.resolve(__dirname, './fixtures')
describe.skipIf(CI.isCI)('index', () => {
  it('createOrUpdateProjectConfig', async () => {
    const name = 'noProjectConfig'
    const p1 = path.resolve(fixturesDir, name, 'project.config.json')
    await createOrUpdateProjectConfig({ root: path.resolve(fixturesDir, name), dest: p1 })
    expect(fs.existsSync(p1)).toBe(true)
  })

  it.each(['vite-native', 'vite-native-skyline', 'vite-native-ts', 'vite-native-ts-skyline'])('%s', async (name) => {
    const root = path.resolve(appsDir, name)
    const p0 = path.resolve(fixturesDir, name, 'package0.json')
    await createOrUpdatePackageJson({ root, dest: p0, command: 'weapp-vite' })
    const p1 = path.resolve(fixturesDir, name, 'project.config.json')
    await createOrUpdateProjectConfig({ root, dest: p1 })
    expect(fs.existsSync(p0)).toBe(true)
    expect(fs.existsSync(p1)).toBe(true)
  })

  it.each(['vite-native', 'vite-native-skyline', 'vite-native-ts', 'vite-native-ts-skyline'])('%s no pkg.json', async (name) => {
    const root = path.resolve(appsDir, name)
    const p0 = path.resolve(fixturesDir, name, 'package.none.json')
    await createOrUpdatePackageJson({ root, dest: p0, command: 'weapp-vite', filename: 'pkg.json' })
    expect(fs.existsSync(p0)).toBe(true)
  })

  it.each(['vite-native', 'vite-native-skyline', 'vite-native-ts', 'vite-native-ts-skyline'])('%s callback', async (name) => {
    const root = path.resolve(appsDir, name)
    const p0 = path.resolve(fixturesDir, name, 'package1.json')
    const res0 = await createOrUpdatePackageJson({
      root,
      dest: p0,
      command: 'weapp-vite',
      write: false,
      cb(set) {
        set('type', 'module')
      },
    })
    const item = omit(res0, ['devDependencies.weapp-vite'])
    expect(item).toMatchSnapshot()
    const p1 = path.resolve(fixturesDir, name, 'project0.config.json')
    const res1 = await createOrUpdateProjectConfig({ root, dest: p1, write: false })
    expect(res1).toMatchSnapshot()
    expect(fs.existsSync(p0)).toBe(false)
    expect(fs.existsSync(p1)).toBe(false)
  })

  it.each(['vite-native', 'vite-native-skyline', 'vite-native-ts', 'vite-native-ts-skyline'])('%s vite.config.ts', async (name) => {
    const root = path.resolve(appsDir, name)
    const res0 = await initViteConfigFile({
      root,
      write: false,
    })
    expect(res0).toMatchSnapshot()
  })

  it.skip.each(['vite-native', 'vite-native-skyline', 'vite-native-ts', 'vite-native-ts-skyline', 'cjs', 'no-pkg-json'])('%s vite.config.ts', async (name) => {
    const root = path.resolve(fixturesDir, name)
    const res = await initConfig({
      root,
      command: 'weapp-vite',
    })
    expect(res).toBeTruthy()
  })

  it.skip.each(['initConfig', 'fullInit'])('%s initConfig', async (name) => {
    const root = path.resolve(fixturesDir, name)
    const res = await initConfig({
      root,
      command: 'weapp-vite',
    })
    expect(res).toBeTruthy()
  })
})
