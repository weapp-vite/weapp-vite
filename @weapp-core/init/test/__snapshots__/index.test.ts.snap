// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > vite-native callback 1`] = `
{
  "//": {
    "cos-wx-sdk-v5": "^1.7.2",
  },
  "author": "ice breaker <1324318532@qq.com>",
  "bugs": {
    "url": "https://github.com/weapp-vite/weapp-vite/issues",
  },
  "dependencies": {
    "@vant/weapp": "^1.11.7",
    "class-variance-authority": "^0.7.1",
    "clsx": "^2.1.1",
    "lodash": "^4.17.21",
    "mini-stores": "^2.3.0",
    "miniprogram-computed": "^7.1.0",
    "tdesign-miniprogram": "^1.11.0",
  },
  "description": "",
  "devDependencies": {
    "@vue/language-plugin-pug": "^3.1.2",
    "@weapp-core/shared": "workspace:*",
    "@weapp-tailwindcss/merge": "1.3.0",
    "autoprefixer": "^10.4.21",
    "cos-wx-sdk-v5": "^1.8.0",
    "dayjs": "^1.11.18",
    "magic-string": "^0.30.21",
    "miniprogram-api-typings": "^4.1.0",
    "miniprogram-simulate": "^1.6.1",
    "postcss": "^8.5.6",
    "sass-embedded": "^1.93.2",
    "sharp": "^0.34.4",
    "svgo": "^4.0.0",
    "tailwindcss": "^3.4.18",
    "typescript": "^5.9.3",
    "vite-imagetools": "^9.0.0",
    "vite-plugin-image-optimizer": "^2.0.2",
    "vite-plugin-inspect": "^11.3.3",
    "weapp-tailwindcss": "^4.6.3",
  },
  "keywords": [],
  "license": "ISC",
  "main": "index.js",
  "name": "vite-native",
  "private": true,
  "repository": {
    "directory": "apps/vite-native",
    "type": "git",
    "url": "git+https://github.com/weapp-vite/weapp-vite.git",
  },
  "scripts": {
    "build": "weapp-vite build",
    "build-npm": "weapp-vite build-npm",
    "build:debug": "cross-env DEBUG=weapp-vite:* weapp-vite build",
    "build:open": "weapp-vite build --open",
    "dev": "weapp-vite dev",
    "dev:0": "weapp-vite dev -m x",
    "dev:1": "weapp-vite dev --mode xx",
    "dev:debug": "cross-env DEBUG=weapp-vite:* node --inspect-brk node_modules/weapp-vite/bin/weapp-vite.js dev",
    "dev:debug:load-entry": "cross-env DEBUG=weapp-vite:load-entry weapp-vite dev -o",
    "dev:open": "weapp-vite dev -o",
    "dev:prod": "weapp-vite dev -o --mode prod",
    "extract": "tsx scripts/extract.ts",
    "g": "weapp-vite generate",
    "hmr": "tsx ./touch.ts",
    "import-json": "tsx scripts/import-json.ts",
    "inspect": "vite preview --outDir .vite-inspect",
    "npm": "weapp-vite npm",
    "open": "weapp-vite open",
    "postinstall": "weapp-tw patch",
    "wxs": "tsup",
  },
  "type": "module",
  "version": "1.0.3",
}
`;

exports[`index > vite-native callback 2`] = `
{
  "appid": "wx6ffee4673b257014",
  "compileType": "miniprogram",
  "condition": {},
  "editorSetting": {
    "tabIndent": "auto",
    "tabSize": 2,
  },
  "libVersion": "trial",
  "miniprogramRoot": "dist/",
  "packOptions": {
    "ignore": [],
    "include": [],
  },
  "setting": {
    "babelSetting": {
      "disablePlugins": [],
      "ignore": [],
      "outputPath": "",
    },
    "coverView": true,
    "enhance": true,
    "es6": true,
    "ignoreUploadUnusedFiles": true,
    "minified": true,
    "packNpmManually": true,
    "packNpmRelationList": [
      {
        "miniprogramNpmDistDir": "./dist",
        "packageJsonPath": "./package.json",
      },
    ],
    "postcss": true,
    "showShadowRootInWxmlPanel": true,
  },
  "srcMiniprogramRoot": "dist/",
}
`;

exports[`index > vite-native vite.config.ts 1`] = `
"import { defineConfig } from 'weapp-vite/config'

export default defineConfig({
  weapp: {
    // weapp-vite options
  },
})
"
`;

exports[`index > vite-native-skyline callback 1`] = `
{
  "author": "ice breaker <1324318532@qq.com>",
  "bugs": {
    "url": "https://github.com/weapp-vite/weapp-vite/issues",
  },
  "description": "",
  "devDependencies": {
    "@vant/weapp": "^1.11.7",
    "autoprefixer": "^10.4.21",
    "miniprogram-api-typings": "^4.1.0",
    "postcss": "^8.5.6",
    "tailwindcss": "^3.4.18",
    "typescript": "^5.9.2",
    "weapp-tailwindcss": "^4.6.3",
  },
  "keywords": [],
  "license": "ISC",
  "main": "index.js",
  "name": "vite-native-skyline",
  "private": true,
  "repository": {
    "directory": "apps/vite-native-skyline",
    "type": "git",
    "url": "git+https://github.com/weapp-vite/weapp-vite.git",
  },
  "scripts": {
    "build": "weapp-vite build",
    "build-npm": "weapp-vite build-npm",
    "dev": "weapp-vite dev",
    "dev:open": "weapp-vite dev -o",
    "g": "weapp-vite generate",
    "open": "weapp-vite open",
    "postinstall": "weapp-tw patch",
  },
  "type": "module",
  "version": "1.0.2",
}
`;

exports[`index > vite-native-skyline callback 2`] = `
{
  "appid": "wx6ffee4673b257014",
  "compileType": "miniprogram",
  "condition": {},
  "editorSetting": {
    "tabIndent": "auto",
    "tabSize": 2,
  },
  "libVersion": "trial",
  "miniprogramRoot": "dist/",
  "packOptions": {
    "ignore": [],
    "include": [],
  },
  "setting": {
    "babelSetting": {
      "disablePlugins": [],
      "ignore": [],
      "outputPath": "",
    },
    "coverView": true,
    "enhance": true,
    "es6": true,
    "minified": true,
    "packNpmManually": true,
    "packNpmRelationList": [
      {
        "miniprogramNpmDistDir": "./dist",
        "packageJsonPath": "./package.json",
      },
    ],
    "postcss": true,
    "showShadowRootInWxmlPanel": true,
  },
  "srcMiniprogramRoot": "dist/",
}
`;

exports[`index > vite-native-skyline vite.config.ts 1`] = `
"import { defineConfig } from 'weapp-vite/config'

export default defineConfig({
  weapp: {
    // weapp-vite options
  },
})
"
`;

exports[`index > vite-native-ts callback 1`] = `
{
  "author": "ice breaker <1324318532@qq.com>",
  "bugs": {
    "url": "https://github.com/weapp-vite/weapp-vite/issues",
  },
  "dependencies": {
    "gm-crypto": "^0.1.12",
  },
  "description": "",
  "devDependencies": {
    "miniprogram-api-typings": "^4.1.0",
    "typescript": "^5.9.2",
  },
  "keywords": [],
  "license": "",
  "name": "vite-native-ts",
  "private": true,
  "repository": {
    "directory": "apps/vite-native-ts",
    "type": "git",
    "url": "git+https://github.com/weapp-vite/weapp-vite.git",
  },
  "scripts": {
    "build": "weapp-vite build",
    "build-npm": "weapp-vite build-npm",
    "dev": "weapp-vite dev",
    "dev:open": "weapp-vite dev -o",
    "g": "weapp-vite generate",
    "open": "weapp-vite open",
  },
  "type": "module",
  "version": "1.0.2",
}
`;

exports[`index > vite-native-ts callback 2`] = `
{
  "appid": "wx6ffee4673b257014",
  "compileType": "miniprogram",
  "condition": {},
  "description": "项目配置文件",
  "editorSetting": {
    "tabIndent": "insertSpaces",
    "tabSize": 2,
  },
  "libVersion": "trial",
  "miniprogramRoot": "dist/",
  "packOptions": {
    "ignore": [],
    "include": [],
  },
  "setting": {
    "babelSetting": {
      "disablePlugins": [],
      "ignore": [],
      "outputPath": "",
    },
    "compileHotReLoad": false,
    "coverView": false,
    "enhance": false,
    "es6": true,
    "ignoreUploadUnusedFiles": true,
    "minified": false,
    "packNpmManually": true,
    "packNpmRelationList": [
      {
        "miniprogramNpmDistDir": "./dist",
        "packageJsonPath": "./package.json",
      },
    ],
    "postcss": false,
    "showShadowRootInWxmlPanel": false,
    "skylineRenderEnable": true,
    "useCompilerPlugins": [
      "typescript",
    ],
  },
  "simulatorPluginLibVersion": {},
  "simulatorType": "wechat",
  "srcMiniprogramRoot": "dist/",
}
`;

exports[`index > vite-native-ts vite.config.ts 1`] = `
"import { defineConfig } from 'weapp-vite/config'

export default defineConfig({
  weapp: {
    // weapp-vite options
  },
})
"
`;

exports[`index > vite-native-ts-skyline callback 1`] = `
{
  "author": "ice breaker <1324318532@qq.com>",
  "bugs": {
    "url": "https://github.com/weapp-vite/weapp-vite/issues",
  },
  "dependencies": {
    "tdesign-miniprogram": "^1.11.0",
  },
  "description": "",
  "devDependencies": {
    "autoprefixer": "^10.4.21",
    "miniprogram-api-typings": "^4.1.0",
    "postcss": "^8.5.6",
    "tailwindcss": "^3.4.18",
    "typescript": "^5.9.2",
    "weapp-tailwindcss": "^4.6.3",
  },
  "keywords": [],
  "license": "",
  "name": "vite-native-ts-skyline",
  "private": true,
  "repository": {
    "directory": "apps/vite-native-ts-skyline",
    "type": "git",
    "url": "git+https://github.com/weapp-vite/weapp-vite.git",
  },
  "scripts": {
    "build": "weapp-vite build",
    "build-npm": "weapp-vite build-npm",
    "dev": "weapp-vite dev",
    "dev:open": "weapp-vite dev -o",
    "g": "weapp-vite generate",
    "open": "weapp-vite open",
    "postinstall": "weapp-tw patch",
  },
  "type": "module",
  "version": "1.0.2",
}
`;

exports[`index > vite-native-ts-skyline callback 2`] = `
{
  "appid": "wx6ffee4673b257014",
  "compileType": "miniprogram",
  "condition": {},
  "description": "项目配置文件",
  "editorSetting": {
    "tabIndent": "insertSpaces",
    "tabSize": 2,
  },
  "libVersion": "trial",
  "miniprogramRoot": "dist/",
  "packOptions": {
    "ignore": [],
    "include": [],
  },
  "setting": {
    "babelSetting": {
      "disablePlugins": [],
      "ignore": [],
      "outputPath": "",
    },
    "compileHotReLoad": false,
    "coverView": false,
    "enhance": true,
    "es6": true,
    "ignoreUploadUnusedFiles": true,
    "minified": false,
    "packNpmManually": true,
    "packNpmRelationList": [
      {
        "miniprogramNpmDistDir": "./dist",
        "packageJsonPath": "./package.json",
      },
    ],
    "postcss": false,
    "showShadowRootInWxmlPanel": false,
    "skylineRenderEnable": true,
    "useCompilerPlugins": [
      "typescript",
    ],
  },
  "simulatorPluginLibVersion": {},
  "simulatorType": "wechat",
  "srcMiniprogramRoot": "dist/",
}
`;

exports[`index > vite-native-ts-skyline vite.config.ts 1`] = `
"import { defineConfig } from 'weapp-vite/config'

export default defineConfig({
  weapp: {
    // weapp-vite options
  },
})
"
`;
